### ISSUE TITLE
watches the wrong build

### ISSUE THREAD

**Issue by nikmartin** [fastlane/watchbuild#5]
Each time I run watchbuild, I get a status in the terminal like:
`Waiting 12 seconds for iTunes Connect to process the build 2.0.8 (2.0.8)... this might take a while...`
which is the build I just uploaded.  A minute or so later, I get the notification on my desktop:
`Version 1.1.18 finished...` which is an older build. The current build is still processing.

**Comment by joshdholtz** [comment]
@nikmartin I am not able to replicate this at all 😕 Is this still an issue for you?

**Comment by nikmartin** [comment]
It is, and here's the issue: the versioning in this project is out of order. There are earlier versions of an app of the same name in this clients account, with later version numbers, because the developer at the time didn't understand or use semver. So my current builds are a lower semver than older versions. Once i skipped ahead to later version numbers, the issue went away.

**Comment by joshdholtz** [comment]
@nikmartin Ahhh, got it! I think I know the issue now 😊 I will look into 👌

**Comment by andrewgarn** [comment]
Seeing this issue as well. Is there a way to get watchbuild to look for a specific version? 

Its picking a version uploaded years ago instead of the latest.

**Comment by joshdholtz** [comment]
@andrewgarn Yes yes, we can make that happen! 

Quick question though... did you update to the latest version that was released earlier this week?

**Comment by andrewgarn** [comment]
Yes, 0.5 fixed the auth issue we were having previously. Thanks for that!

**Comment by rogerluan** [comment]
Awesome! Thanks for confirming @andrewgarn 🤗 

I'm closing this issue then. Let us know if you see any other issues after upgrading to v0.5.0 🙏 

Cheers!

**Comment by andrewgarn** [comment]
To clarify, it fixed the auth issue in #17 but i'm still seeing it pick the wrong build after updating to 0.5.0

**Comment by rogerluan** [comment]
Ah got it! Sorry about that!
Reopening

**Comment by rogerluan** [comment]
@andrewgarn are you facing this issue? https://github.com/fastlane/watchbuild/issues/5#issuecomment-379752191

**Comment by andrewgarn** [comment]
> @andrewgarn are you facing this issue? [#5 (comment)](https://github.com/fastlane/watchbuild/issues/5#issuecomment-379752191)

Still an issue as of January 2022. There is an old build from **years ago** watchbuild picks instead of the one uploaded today. The version number (2.9) is also much earlier than the current version (9.3).

Checking App Store connect the very old (2.9) build seems to be stuck in **Processing** state. There is no way to cancel, expire or delete this build in the App Store Connect website.

Can we have watchbuild prioritise newer builds, or allow a version into watchbuild alongside bundle identifier?

Thanks

**Comment by ryzn0518** [comment]
@nikmartin 
Since we upload appstore through the Private Key, how do we operate which user to use?

**Comment by nikmartin** [comment]
> @nikmartin 
> Since we upload appstore through the Private Key, how do we operate which user to use?
> 

I have no idea. You may want to open a new issue, it doesn't seem related.
